### PR TITLE
Modifies ValitRuleStringExtensions to consider a zero length string a valid string.

### PR DESCRIPTION
**Pull request by bovorasr**
# Info
This Pull Request is related to issue no. [116](https://github.com/valit-stack/Valit/issues/166) 

# Changes
- Changed ValitRuleStringExtensions precondition checks from String.IsNullOrEmpty to only check null.
    - Causes MinValue(0) against "" to pass validation (previously would fail validation).
    - Causes MaxValue(m), where m > 0 to pass validation (previously would fail validation).
    - In general, allows the empty string to be a valid value for a string.

### CODE REVIEW

File: tests/Valit.Tests/Strategies/FailFast.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ public void FailFast_Strategy_Checks_Until_First_Error_While_Error_Is_Not_First(
 
             result.Succeeded.ShouldBe(false);
             result.ErrorMessages.Count().ShouldBe(1);
-            result.ErrorMessages.ShouldContain(M2);
+            result.ErrorMessages.ShouldContain(M3);
```
Comment (bovorasr) [inline tests/Valit.Tests/Strategies/FailFast.cs:71]: M2 == "Field is required". M3 == "This value should be at least N long." 

Same for any other changes similar to this in tests.

File: tests/Valit.Tests/String/String_IsEqualTo_Tests.cs [hunk 2]
```diff
@@ -36,7 +36,7 @@ public void String_IsEqualTo_Returns_Proper_Result_For_Left_Value(string value,
         [Theory]
         [InlineData("text", false)]
         [InlineData("other", false)]
-        [InlineData("", false)]
+        [InlineData("", true)]
```
Comment (bovorasr) [inline tests/Valit.Tests/String/String_IsEqualTo_Tests.cs:39]: This seems to make more sense now: "" is now considered equal to "". Previously the test was saying comparing "" to "" should result in false.

File: tests/Valit.Tests/String/String_MaxLength_Tests.cs [hunk 3]
```diff
@@ -33,9 +33,9 @@ public void String_MaxLength_Returns_Proper_Result_For_Left_Value(int value, boo
         }
 
         [Theory]
-        [InlineData(0, false)]
-        [InlineData(4, false)]
-        [InlineData(8, false)]
+        [InlineData(0, true)]
+        [InlineData(4, true)]
+        [InlineData(8, true)]
```
Comment (bovorasr) [inline tests/Valit.Tests/String/String_MaxLength_Tests.cs:38]: A string of zero length now considered to have a length less than or equal to 0, 4, or 8.

File: tests/Valit.Tests/String/String_MinLength_Tests.cs [hunk 4]
```diff
@@ -33,7 +33,7 @@ public void String_MinLength_Returns_Proper_Result_For_Left_Value(int value, boo
         }
 
         [Theory]
-        [InlineData(0, false)]
+        [InlineData(0, true)]
```
Comment (bovorasr) [inline tests/Valit.Tests/String/String_MinLength_Tests.cs:36]: A string of zero length is now considered to be at least zero characters long. The others, 4 & 8 remain false.

File: tests/Valit.Tests/String/String_Required_Tests.cs [hunk 5]
```diff
@@ -42,7 +42,7 @@ public void String_Required_Fails_For_Empty_Value()
                 .For(_model)
                 .Validate();
 
-            result.Succeeded.ShouldBeFalse();
+            result.Succeeded.ShouldBeTrue();
```
Comment (bovorasr) [inline tests/Valit.Tests/String/String_Required_Tests.cs:45]: An empty string is considered a valid value of a string.

**Review by GooRiOn — APPROVED** [review]
LGTM 👍  I'm not sure yet, whether I suppose to bump major number of the project (breaking change) or feature number. Anyway thank you for helping with this issue, and... welcome aboard :)

**Pull request merged**